server/src/main/java/io/unitycatalog/server/service/TemporaryVolumeCredentialsService.java

```diff
@@ -51,4 +64,34 @@ private Set<CredentialContext.Privilege> volumeOperationToPrivileges(VolumeOpera
       case UNKNOWN_VOLUME_OPERATION -> Collections.emptySet();
     };
   }
+
+  private void authorizeForOperation(GenerateTemporaryVolumeCredential generateTemporaryVolumeCredential) {
+
+    // TODO: This is a short term solution to conditional expression evaluation based on additional request parameters.
+    // This should be replaced with more direct annotations and syntax in the future.
+
+    String readExpression = """
+          #authorizeAny(#principal, #schema, OWNER, USE_SCHEMA) && #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #volume, OWNER, READ_VOLUME)
+          """;
+
+    // TODO: add WRITE_VOLUME to the expression
+    String writeExpression = """
+          #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) &&
+          #authorizeAny(#principal, #schema, OWNER, USE_SCHEMA) &&
+          (#authorize(#principal, #schema, OWNER) || #authorizeAll(#principal, #volume, READ_VOLUME))
```

Comment:
```suggestion
          #authorize(#principal, #schema, OWNER)
```

Reply:
This should be `#volume` not `#schema` right?